com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java

```diff
@@ -64,7 +76,21 @@ private ErrorType getSubProcessErrorType(HistoricProcessInstance subProcess) {
     protected abstract ExecutionStatus onAbort(DelegateExecution context, ErrorType errorType) throws MonitoringException;
 
     protected ExecutionStatus onSuccess(DelegateExecution context) {
+        duplicateProcessVariablesFromSubProcess(context);
         return ExecutionStatus.SUCCESS;
     }
 
+    private void duplicateProcessVariablesFromSubProcess(DelegateExecution context) {
+        String subProcessId = StepsUtil.getSubProcessId(context);
+        List<String> processVariablesToDuplicate = getProcessVariablesToDuplicate();
+        for (String processVariable : processVariablesToDuplicate) {
+            HistoricVariableInstance historicVariableInstance = activitiFacade.getHistoricVariableInstance(subProcessId, processVariable);
+            if (historicVariableInstance != null) {
+                context.setVariable(processVariable, historicVariableInstance.getValue());
+            }
+        }
+    }
+
+    protected abstract List<String> getProcessVariablesToDuplicate();

```

Comment:
The same here... processVariablesToDuplicate is not talking much in the Activiti context

Reply:
Done